test/js_test.go

```diff
@@ -955,6 +955,250 @@ func TestJetStreamSubscribe(t *testing.T) {
 	}
 }
 
+func TestPullSubscribeFetchChan(t *testing.T) {
+	s := RunBasicJetStreamServer()
+	defer shutdownJSServerAndRemoveStorage(t, s)
+
+	nc, js := jsClient(t, s)
+	defer nc.Close()
+
+	_, err := js.AddStream(&nats.StreamConfig{
+		Name:     "TEST",
+		Subjects: []string{"foo"},
+	})
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	t.Run("basic fetch", func(t *testing.T) {
```

Comment:
need a test where the batch is delivered all the messages, after all have been delivered then try to get the delivered messages